csrc/python_frontend/fusion_record.h

```diff
@@ -1986,6 +1986,60 @@ struct TorchGatherOpRecord : RecordFunctor {
   int64_t dim_;
 };
 
+//! Similar to TorchGatherOpRecord but enforces that non-index dimension
+//! extents match between index tensor and value tensor.
+struct TakeAlongAxisOpRecord : RecordFunctor {
+  TakeAlongAxisOpRecord(
+      std::vector<State> _args,
+      std::vector<State> _outputs,
+      int64_t dim)
+      : RecordFunctor(
+            std::move(_args),
+            std::move(_outputs),
+            "ops.take_along_axis",
+            serde::RecordType_TakeAlongAxisOp),
+        dim_(dim) {}
```

Comment:
nitpick: can we add check here on input/arg1 with matching dimension? as well as `dim` within range and things like that.
Just so error pops up earlier at the time when we call the function, rather than later when exiting fusion definition context.

Reply:
Good idea to add dim check. Will add it to gather too while I'm here.